packages/react/src/components/dropdown-list/dropdown-list.tsx

```diff
@@ -204,426 +140,258 @@ export const DropdownList: VoidFunctionComponent<DropdownListProps> = ({
-    const wrapperRef = useRef<HTMLDivElement>(null);
+    const textboxRef = useRef<HTMLDivElement>(null);
     const listboxRef = useRef<HTMLDivElement>(null);
-    const [shouldFilterOptions, setShouldFilterOptions] = useState<boolean | undefined>(searchable);
-    const filteredOptions = useMemo(
-        () => {
-            if (!shouldFilterOptions) {
-                return options;
-            }
-            return filterOptions(options, searchValue);
-        },
-        [options, searchValue, shouldFilterOptions],
-    );
 
-    const findOptionByValue: (needle?: string) => Option | undefined = useCallback(
-        (needle) => options.find((option) => option.value === needle),
-        [options],
-    );
-    const [inputValue, setInputValue] = useState(() => findOptionByValue(selectedOptionValue)?.label ?? '');
-
-    useEffect(() => {
-        const wantedOption = findOptionByValue(value);
-        if (wantedOption) {
-            setSelectedOptionValue(wantedOption.value);
-            setInputValue(wantedOption.label);
-        } else if (skipOption && skipOption.value === value) {
-            setSelectedOptionValue(skipOption.value);
-            setInputValue('');
-        }
-    }, [findOptionByValue, options, skipOption, skipSelected, value]);
-
-    const handleOpen: () => void = useCallback(() => {
-        if (!disabled) {
-            if (!open) {
-                setFocus(true);
-                if (searchable && inputRef.current) {
-                    inputRef.current.focus();
-                    setFocusedValue('');
-                } else {
-                    setFocusedValue(selectedOptionValue);
-                }
-            } else {
-                inputRef.current?.focus();
-                setAutofocus(false);
-                setFocusedValue('');
-            }
-            setOpen(!open);
-        }
-    }, [disabled, open, searchable, selectedOptionValue]);
-
-    const handleClickOutside: (event: MouseEvent) => void = useCallback((event) => {
-        const clickIsOutside = !eventIsInside(event, wrapperRef.current, listboxRef.current);
-        const shouldClose = (wrapperRef.current === null || clickIsOutside) && open;
-
-        if (shouldClose) {
-            handleOpen();
-            inputRef.current?.blur();
-        }
-    }, [open, handleOpen]);
-
-    useEffect(() => {
-        document.addEventListener('mouseup', handleClickOutside);
-        return () => document.removeEventListener('mouseup', handleClickOutside);
-    }, [handleClickOutside, open]);
+    const [open, setOpen] = useState(defaultOpen);
 
-    function findOption(optionsArray: Option[], findValue: string): Option | undefined {
-        return optionsArray.find((option) => option.label.toLowerCase() === findValue.toLowerCase());
+    function findOptionByValue(searchValue?: string): DropdownListOption | undefined {
+        return options.find((option) => option.value === searchValue);
```

Comment:
On devrait ajouter ça? J'ai l'impression que quelqu'un pourrait passer une value qui est disabled en prop.
```suggestion
        return options.find((option) => option.value === searchValue && !option.disabled);
```

Reply:
Je l'ai mentionné aussi dans la PR du listbox, est-ce qu'on permet un comportement similaire à ce qu'on peut faire avec un `<option disabled selected>`? L'option n'est pas sélectionnable par l'utilisateur mais elle peut l'être si le component est controlled ou en passant une `defaultValue`. Qu'en penses-tu?

Reply:
Hmm, on est effectivement sûrement mieux de se coller à ce que `option` permet pour l'instant. On révisitera au besoin.